core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+<script type="text/ng-template" id="components/topNavigationBar">
```

Comment:
Maybe it would be simpler if this directive were split up. There's a major division on how the directive looks and behaves based on whether the current user is logged in. That might be a good start for a separation. There are probably a few other splits possible, too.